github/misc_test.go

```diff
@@ -168,3 +168,48 @@ func TestRepositoriesService_ListServiceHooks(t *testing.T) {
 		t.Errorf("Repositories.ListServiceHooks returned %+v, want %+v", hooks, want)
 	}
 }
+
+func TestParseURL(t *testing.T) {
+	const (
+		owner, repo = "bazelbuild", "bazel"
+	)
+	tests := []struct {
+		url                 string
+		wantOwner, wantRepo string
+		wantErr             bool
+	}{
+		{url: "https://github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "http://github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "https://www.github.com/bazelbuild/bazel/tree/master", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/releases/tag/0.2.1", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/commit/19b...edc", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/archive/0.2.1.zip", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel/", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/bazel", wantOwner: owner, wantRepo: repo},
+		{url: "https://github.com/bazelbuild/", wantOwner: owner},
+		{url: "https://github.com/bazelbuild", wantOwner: owner},
+		{url: "https://mygithub.com/bazelbuild", wantErr: true},
+		{url: "https://github.com///", wantErr: true},
```

Comment:
I don't know if you want to handle this case, but the following URL actually works:

https://github.com/////google/////go-github

If you do want to handle it, it can probably be done by doing `parts := strings.SplitN(path.Clean(u.Path), "/", 4)`.